common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/WebViewAuthorizationFragment.java

```diff
@@ -227,9 +237,113 @@ public boolean onTouch(final View view, final MotionEvent event) {
+    /**
+     * Call this method to grant the permission to access the camera resource.
+     * The granted permission is only valid for the current WebView.
+     */
+    private void acceptCameraRequest() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            final String[] cameraPermission = new String[] {
+                    PermissionRequest.RESOURCE_VIDEO_CAPTURE
+            };
+            if (mCameraPermissionRequest != null) {
+                mCameraPermissionRequest.grant(cameraPermission);
+            }
+        }
+    }
+
+    /**
+     * Call this method to deny the permission to access the camera resource.
+     */
+    private void denyCameraRequest() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP &&
+                mCameraPermissionRequest != null) {
+            mCameraPermissionRequest.deny();
+        }
+    }
+
+    /**
+     * Determines whatever if the camera permission has been granted.
+     *
+     * @return true if the camera permission has been granted, false otherwise.
+     */
+    private boolean isCameraPermissionGranted() {
+        return ContextCompat.checkSelfPermission(requireContext(), Manifest.permission.CAMERA)
+                == PackageManager.PERMISSION_GRANTED;
+    }
+
+    /**
+     * Determines whatever if the given permission request is for the camera resource.
+     *
+     * @param request The permission request.
+     * @return true if the given permission request is for camera, false otherwise.
+     */
+    private boolean iSPermissionRequestForCamera(final PermissionRequest request) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            return request.getResources().length == 1 &&
+                    PermissionRequest.RESOURCE_VIDEO_CAPTURE.equals(request.getResources()[0]);
+        }
+        return false;
+    }
+
+    private final ActivityResultLauncher<String> cameraRequestActivity = registerForActivityResult(
+            new ActivityResultContracts.RequestPermission(),
+            permissionGranted   -> {
+                if (permissionGranted) {
+                    acceptCameraRequest();
+                }
+                else {
+                    denyCameraRequest();
+                }
+            }
+    );
+
+    /**
+     * Launches the camera permission request for the app.
+     */
+    private void requestCameraPermission() {
+        cameraRequestActivity.launch(Manifest.permission.CAMERA);
+    }
+
+    /**
+     * Shows a dialog to the user explaining why the camera permission is required.
+     * If the user accepts the dialog, the camera permission request will be launched.
+     * If the user denies the dialog, the camera permission request will be denied.
+     */
+    private void showCameraRationale() {
+        final AlertDialog.Builder builder = new AlertDialog.Builder(getContext());
+        builder.setMessage("The app requires the camera permission to scan the QR Code, " +
```

Comment:
This would need to be localized.

Reply:
no, checking this right now.

Reply:
https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/2258/commits/3c1aad63823f6e287848a3972d69b76a3f833e1b
https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/2258/commits/26649bba7a24df0159f9bdc8d0291944b5352971